plugins/components/position3d/Position3d.hpp

```diff
@@ -0,0 +1,181 @@
+    #include <libconfig.h++>
+    #include <iostream>
+    #include <iomanip>
+
+namespace Components {
+    /**
+     * @brief Position3d component class in Components.
+     * 
+     * This class represents the position of an entity, storing the x and y coordinates,
+     * as well as a z to manage the entity's depth in the rendering order. It also provides
+     * methods to serialize and deserialize the position data.
+     */
+    class Position3d : public AComponent<Position3d> {
+    public:
+        /**
+         * @brief Default constructor for the Position3d component.
+         * 
+         * Initializes the position components (x, y, z) to zero.
+         */
+        Position3d() : AComponent("Position3d"), x(0), y(0), z(0), floatX(0), floatY(0), floatZ(0) {};
+        Position3d(libconfig::Setting &config) : AComponent("Position3d")
+        {
+            if (!config.lookupValue("x", x)) x = 0;
+            if (!config.lookupValue("y", y)) y = 0;
+            if (!config.lookupValue("z", z)) z = 1;
+        }
+
+        /**
+         * @brief Parameterized constructor for the Position3d component.
+         * 
+         * Initializes the position with the provided x and y coordinates and defaults the z to zero.
+         * 
+         * @param x The X coordinate of the position.
+         * @param y The Y coordinate of the position.
+         */
+        Position3d(int32_t x, int32_t y, int32_t z) : AComponent("Position3d"), x(x), y(y), z(z), floatX(0), floatY(0), floatZ(0) {};
+
+        /**
+         * @brief Default destructor for the Position3d component.
+         */
+        ~Position3d() = default;
+
+        /**
+         * @brief Serializes the position components into a byte vector.
+         * 
+         * Converts the x, y, and z components into network byte order for transmission or storage.
+         * 
+         * @return A vector of bytes representing the serialized position components.
+         */
+        std::vector<uint8_t> serialize() override {
+            __network.x = htonl(x);
+            __network.y = htonl(y);
+            __network.z = htonl(z);
+            return std::vector<uint8_t>(__data, __data + sizeof(__data));
+        };
+
+        /**
+         * @brief Deserializes the position components from the provided byte vector.
+         * 
+         * Reads the x, y, and z components in network byte order.
+         * 
+         * @param data A vector of bytes representing the serialized position components.
+         * @throws std::runtime_error If the data size is invalid.
+         */
+        void deserialize(std::vector<uint8_t> &data) override {
+            if (data.size() != sizeof(__data))
+                throw std::runtime_error("Invalid data size for position component");
+            x = ntohl(*reinterpret_cast<int32_t *>(data.data()));
+            y = ntohl(*reinterpret_cast<int32_t *>(data.data() + 4));
+            z = ntohl(*reinterpret_cast<int32_t *>(data.data() + 8));
```

Comment:
is `int32_t` really what is needed ? If not you could change this to `uint32_t`

Reply:
Yes it is needed since we can have negative values in a 3d world (also in a 2d world but yeah, we didnt use that)